src/export/InstanceExporter.ts

```diff
@@ -447,6 +455,74 @@ export class InstanceExporter implements Fishable {
     }
   }
 
+  /**
+   * This function analyzes an exported instance looking at all sliced paths and attempting to
+   * detect cases where an array item matches a required (implied) slice but was not assigned
+   * using a slice name. We want to detect and warn on this because it very often results in
+   * undesirable outputs like repeated and/or incomplete array items. Before manualSliceOrdering,
+   * most of these cases just overwrote the implied slice instead of being prepended before the
+   * slice.
+   */
+  private checkForNamelessSlices(
+    fshDef: Instance,
+    instanceDef: InstanceDefinition,
+    instanceOfStructureDefinition: StructureDefinition
+  ) {
+    const nonChoiceSlicedElements = instanceOfStructureDefinition.elements.filter(
+      e => e.slicing && !e.path.endsWith('[x]')
+    );
+    for (const slicingEl of nonChoiceSlicedElements) {
+      // Use the sliced element's path to get all potential element values from the instance
+      const values = collectValuesAtElementIdOrPath(slicingEl.id, instanceDef);
+      // Find the values that do not have named slices
+      const unnamedValues = values.values.filter(v => v != null && v._sliceName == null);
+      if (unnamedValues.length) {
+        // Find the values that do have named slices (for comparison)
+        const namedValues = values.values
+          .filter(v => v?._sliceName != null)
+          .map(v => {
+            const cleanValue = cloneDeep(v);
+            delete cleanValue._sliceName;
+            return { name: v._sliceName, value: cleanValue };
+          });
+        if (namedValues.length) {
+          for (const value of unnamedValues) {
+            // It's a match if the unnamed value is a compatible superset of the named value (i.e., slice)
+            const matches = namedValues
+              .filter(nv => matchesPattern(value, nv.value))
+              .map(nv => nv.name);
+            if (matches.length) {
+              if (namedValues.every(nm => isEqual(value, nm.value))) {
```

Comment:
Good catch. My intent was that if an item exactly matches a slice but _also_ partially matches another slice, I want the second message (because I can't say for sure which slice is the right one). But clearly I messed that up because it fails the use case where there are multiple slices and only one match (that happens to be exact). I'll work on that.

Reply:
Addressed in [4fea2ef](https://github.com/FHIR/sushi/pull/1200/commits/4fea2efe5f1365762afddea29f0838f921ec72b0). Now it works according to my intent: (a) use the exactly message if _all_ matches are exact, (b) use general message when  some matches are not exact.